pkg/reconciler/pipelinerun/resources/pipelinerunstate.go

```diff
@@ -423,7 +423,7 @@ func (facts *PipelineRunFacts) GetSkippedTasks() []v1beta1.SkippedTask {
 			}
 			// include the when expressions only when the finally task was skipped because
 			// its when expressions evaluated to false (not because results variables were missing)
-			if !rprt.PipelineTask.WhenExpressions.HaveVariables() {
+			if rprt.IsFinallySkipped(facts).SkippingReason == WhenExpressionsSkip {
```

Comment:
whoa nice update, much clearer!!

(im a bit confused about why "has variables" is enough - e.g. what if hte when expression had variables but it was skipped for a different reason? anyway that doesnt matter now b/c you've fixed it XD)

Reply:
it wasn't enough but a good proxy for many cases